lib/exception_handling.rb

```diff
@@ -177,26 +177,15 @@ def log_error(exception_or_string, exception_context = '', controller = nil, tre
         exception_info = ExceptionInfo.new(ex, exception_context, timestamp, controller || current_controller, data_callback)
 
         if stub_handler
-          return stub_handler.handle_stub_log_error(exception_info.data)
-        end
-
-        write_exception_to_log(ex, exception_context, timestamp)
-        execute_custom_log_error_callback(exception_info.enhanced_data, exception_info.exception, treat_like_warning)
-
-        if treat_like_warning  #Bail out before sending to honeybadger or email
-          return
-        end
-
-        if !ex.is_a?(Warning)
-          if honeybadger?
-            send_exception_to_honeybadger(exception_info)
-          end
-
-          if should_send_email?
-            log_error_email(exception_info)
-          end
+          stub_handler.handle_stub_log_error(exception_info.data)
+        else
+          write_exception_to_log(ex, exception_context, timestamp)
+          external_notification_results = unless treat_like_warning || ex.is_a?(Warning)
+                                            send_external_notifications(exception_info)
+                                          end || {}
+          execute_custom_log_error_callback(exception_info.enhanced_data, exception_info.exception, treat_like_warning, external_notification_results)
+          nil
```

Comment:
Is this `nil` return value needed?

Reply:
I have a feeling that it is. Previously, `log_error_email` was the last method executed within `log_error`. This method specified a `nil` return value at the end. I believe we generally expect `log_error` to return`nil`. In fact, I think I've seen tests in web which expect that...

Since I've rearranged things, I can imagine that the custom log error callback may return an arbitrary value. So, I believe it is necessary to maintain the current contract.

Reply:
👍